README.md

```diff
@@ -1,51 +1,92 @@
-# nginx_otel
+# NGINX Native OpenTelemetry (OTel) Module
 
-This project provides support for OpenTelemetry distributed tracing in Nginx, offering:
+## What is OpenTelemetry
+OpenTelemetry (OTel) is an observability framework for monitoring, tracing, troubleshooting, and optimizing applications. OTel enables the collection of telemetry data from a deployed application stack.
 
-- Lightweight and high-performance incoming HTTP request tracing
-- [W3C trace context](https://www.w3.org/TR/trace-context/) propagation
-- OTLP/gRPC trace export
-- Fully Dynamic Variable-Based Sampling
+## What is the Native NGINX Otel Module
+The `ngx_otel_module` dynamic module enables NGINX Open-Source or NGINX Plus to send telemetry data to an OTel collector. It provides support for [W3C trace context](https://www.w3.org/TR/trace-context/) propagation, OTLP/gRPC trace exports and offers several benefits over exiting OTel modules, including:
+
+### Better Performance ###
+3rd-party OTel implementations reduce performance of request processing by as much as 50% when tracing is enabled. The NGINX Native module limits this impact to approximately 10-15%.
+
+### Easy Provisioning ###
+Setup and configuration can be done right in NGINX configuration files.
+
+### Fully Dynamic Variable-Based Sampling ###
+The module provides the ability to trace a particular session by cookie/token. [NGINX Plus](https://www.nginx.com/products/nginx/), available as part of a [commercial subscription](https://www.nginx.com/products/), enables dynamic module control via the [NGINX Plus API](http://nginx.org/en/docs/http/ngx_http_api_module.html) and [key-value store](http://nginx.org/en/docs/http/ngx_http_keyval_module.html) modules.
 
 ## Building
+Follow these steps to build the `ngx_otel_module` dynamic module on Ubuntu or Debian based systems:
 
-Install build tools and dependencies:
+Install build tools and dependencies.
 ```bash
-  $ sudo apt install cmake build-essential libssl-dev zlib1g-dev libpcre3-dev
-  $ sudo apt install pkg-config libc-ares-dev libre2-dev # for gRPC
+sudo apt install cmake build-essential libssl-dev zlib1g-dev libpcre3-dev
+sudo apt install pkg-config libc-ares-dev libre2-dev # for gRPC
 ```
 
-Configure Nginx:
+Clone the NGINX repository.
 ```bash
-  $ ./configure --with-compat
+git clone https://github.com/nginx/nginx.git
```

Comment:
This implies building nginx from source, as module will be compatible only with that version of nginx. To build module compatible with prebuilt nginx binary one have to use matching version of nginx source, though we probably should provide nginx-otel packages for OSS as well (if not doing so already).

Reply:
Prior to adding the clone step, it was confusing where the configure script was coming from, so that part seems necessary. Given what you're saying, does it make sense to change the steps to compile and install nginx from source (rather than binary later on in the readme) along with the built otel module? Seems to be the only way to guarantee codebase parity. To my knowledge, we're not yet hosting nginx-otel packages for OSS.

Reply:
'configure' comes from nginx source code, which one can obtain in various ways, e.g. source code releases available via nginx.org contain 'configure' at the top level (some distros might also provide nginx-dev packages). The idea was just to highlight the need for `--with-compat`.
Yes, given this line, I'd remove mention of binaries later on. And, yes, building nginx alongside with the module is the easiest way to guaranty compatibility.

Reply:
https://github.com/nginxinc/nginx-otel/pull/7/commits/b85039b4efea39660ad2378b14a8760771bc1726 addresses this. Please let me know if you see any other areas that need adjusting.